compiler/src/compilers/cpp_compiler.rs

```diff
@@ -0,0 +1,47 @@
+#![allow(unused)]
+
+//use std::env;
+use std::path::PathBuf;
+use std::process::Command;
+
+use super::compiler::Compiler;
+use super::super::data::input_data::InputData;
+use super::super::data::output_data::OutputData;
+use super::super::storage::name_generator::generate_filename;
+
+pub struct CppCompiler {
+
+}
+
+impl Compiler for CppCompiler {
+    fn compile(&self, input_data: &InputData) -> OutputData {
```

Comment:
Does this function always succeed, why not returning `Result<..>`?

Reply:
There can be a lot of option that may bring this function to unsuccessful result. For instance "lack of the specific compiler" mentioned in previous comment. So, it will be more efficient to return "Result<..>" in this case - just added corresponding TODO record.